### ISSUE TITLE
Headers should be case insensitive

### ISSUE THREAD

**Issue by fappelman** [swift-server/async-http-client#185]
I have run into an issue where the library did not follow up on a `301`. This was when I accessed the website from *wired*: [https://www.wired.com/story/the-science-of-this-pandemic-is-moving-at-dangerous-speeds](url) 

It turns out that this website returns the location header as:

```log
location: https://www.wired.com/story/the-science-of-this-pandemic-is-moving-at-dangerous-speeds/
```
Note that the location is written as `location` and not as `Location`.

In the redirect handler this is managed as:

```swift
guard let location = headers.first(where: { $0.name == "Location" }) else {
     return nil
}
```

According to the RFC: [https://ietf.org/rfc/rfc7230.html#section-3.2](url) the headers are case insensitive. Hence I believe this code should become:

```swift
guard let location = headers.first(where: { $0.name.lowercased() == "location" }) else {
     return nil
}
```

I did not check but I assume there this may be true in other areas as well. The above changed fixed it for me.

I  used version `1.1.0` of the package.

**Comment by artemredkin** [comment]
@fabianfett should be fixed now, tagged `1.1.1`

**Comment by fappelman** [comment]
Thanks. Appreciated.
